### DISCUSSION TITLE
What are the limitations of V's hot code reloading mechanism? Any detailed documents?

### DISCUSSION THREAD

**Discussion by lijiansong** [vlang/v#11048] [General]
I find that current implementation of V's hot code reloading is very simple, it's just a wrapper of  the `dl` library.
<https://github.com/vlang/v/tree/master/vlib/v/live> & <https://github.com/vlang/v/tree/master/examples/hot_reload>

So what are the limitations of V's hot code reloading mechanism? Any detailed documents?

**Comment by lijiansong** [comment]
shared library injection with the `dl` library and `ptrace`?

**Comment by nova-nowiz** [comment]
what is this dl and ptrace libraries you are talking about @lijiansong?
and a good thing to test would be changes in data structures ;)

**Reply by spytheman** [reply]
Changes in data structure layout is not supported, only changing functions. If you change the data structure layout, your previous memory state would be invalid, and you will have to restart the live program yourself, losing the current state.

**Reply by nova-nowiz** [reply]
mun-lang is a good language that has hot reloadability baked in for structs: https://docs.mun-lang.org/ch03-04-hot-reloading-structs.html
too bad v doesn't have this :/

**Reply by lijiansong** [reply]
> Changes in data structure layout is not supported, only changing functions. If you change the data structure layout, your previous memory state would be invalid, and you will have to restart the live program yourself, losing the current state.

@spytheman @Narice 
Thanks for reply.
Will these cases be supported:
- changing the signature of the function, for example, changing its args numbers/type or the return type?
- changing the state of current loops, for example, a state info is shared among some subroutines, will the state info hold its original data after the hot reloading?
- changing the state of the global vars, will their global state hold after the hot code reloading?
